spec/cases/with_exception_in_finish.rb

```diff
@@ -0,0 +1,19 @@
+require './spec/cases/helper'
```

Comment:
I kept them separate since they require different logic in both the hook and the workers to ensure consistent behavior